### ISSUE TITLE
Matching specific file is not found when pattern contains parentheses

### ISSUE THREAD

**Issue by zrh122** [mrmlnc/fast-glob#223]
### Environment

  * OS Version: Windows 7 
  * Node.js Version: v10.16.2

### Actual behavior
two results is not same

### Expected behavior
two results should be the same

### Steps to reproduce

create directory and file, and run the code

### Code sample

directory structure:
```
├── tmp(asd)
    └── a.tmp
```
from above directory, run script:
```js
const fg = require('fast-glob');
const paths = fg.sync('./tmp\\(asd\\)/a.tmp');
console.log(paths); // []
const paths1 = fg.sync('./tmp\\(asd\\)/*');
console.log(paths1); // ['./tmp(asd)/a.tmp']
```

**Comment by mrmlnc** [comment]
So, after some research we have a three ways to solve this problem:

* Check that the pattern is a static pattern and then strip escape symbols. In fact, this is a modification of user input, which can lead to problems.
* Check that the pattern has escape symbols and then move him to dynamic patterns. Looks good for me, but it can lead to loss of performance (only for patterns with escape symbols).
* Do not escape symbols for filepath's (non glob pattern's). It is required that the user understood the difference, which is difficult and will still lead to problems.

**Comment by mrmlnc** [comment]
We have a some problem here with pattern `file-\\(suffix\\).md`.

```js
{
  dynamic: true,
  positive: [ 'file-\\(suffix\\).md' ],
  negative: [],
  base: 'file-', // by glob-parent
  patterns: [ 'file-\\(suffix\\).md' ]
}
```

https://github.com/gulpjs/glob-parent/issues/12
https://github.com/gulpjs/glob-parent/blob/d4975484222bb811c02dc2dc268ea48e2fb7ae6d/test/index.test.js#L73

**Comment by mrmlnc** [comment]
Waiting for https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38521

**Comment by mrmlnc** [comment]
Will be fixed with `fast-glob@3.1.0`.
